### DISCUSSION TITLE
StateNotifierProvider becomes legacy as from 2.0?

### DISCUSSION THREAD

**Discussion by vbalagovic** [rrousselGit/riverpod#1927] [Q&A] [answered]
Hi,

in some articles I found that, StateProvider , StateNotifierProvider, ChangeNotifierProvider are becoming legacy in favour of NotifierProvider & AsyncNotifierProvider but somehow i didn't find this in v2 documentation.

As someone who was using StateNotifierProvider in some bigger projects, and should now start new project am currently puzzled if I should stop using them and switch to AsyncNotifierProvider?

Any information or tip is welcome. Thanks!

**Comment by rrousselGit** [comment, marked as answer]
Yes. They are the way forward, although best combined with riverpod_generator.

Without code-generation, they aren't quite perfect to use when combining them with family/autoDispose

**Reply by vbalagovic** [reply]
Thanks for answering :)
